app/Roster/Roster/RosterQueryTrait.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+
+namespace App\Roster;
+
+
+use App\Roster;
+use App\TableProperties;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+
+trait RosterQueryTrait
+{
+
+    public function rosters()
+    {
+        return $this->hasMany(Roster::class, $this->foriegnPivotKey());
+    }
+
+    /**
+     * @return BelongsToMany
+     */
+    public function players()
+    {
+        return $this->rosters()->with('player')->ofPlayerType();
+    }
+
+    /**
+     * @return BelongsToMany
+     */
+    public function proficiencies()
+    {
+        return  $this->rosters()->with('proficiency');
+    }
+
+    public function teams()
```

Comment:
Team Model do not have teams, what it will do in case of teams? Query Trait should be divided accordingly e.g 
RosterTeamsQueryTrait with this teams() method